File: backend-java/test-design-assistant-core/src/main/java/com/sinosoft/testdesign/entity/PromptTemplate.java
package com.sinosoft.testdesign.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 提示词模板实体
 * 
 * @author sinosoft
 * @date 2024-01-01
 */
@Data
@Entity
@Table(name = "prompt_template")
public class PromptTemplate {
    
    /**
     * 主键ID
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    /**
     * 模板编码
     */
    @Column(name = "template_code", unique = true, nullable = false, length = 100)
    private String templateCode;
    
    /**
     * 模板名称
     */
    @Column(name = "template_name", nullable = false, length = 500)
    private String templateName;
    
    /**
     * 模板分类
     */
    @Column(name = "template_category", length = 100)
    private String templateCategory;
    
    /**
     * 提示词类型
     */
    @Column(name = "template_type", length = 50)
    private String templateType;
    
    /**
     * 模板内容
     */
    @Column(name = "template_content", nullable = false, columnDefinition = "TEXT")
    private String templateContent;
    
    /**
     * 模板变量定义（JSON格式）
     */
    @Column(name = "template_variables", columnDefinition = "TEXT")
    private String templateVariables;
    
    /**
     * 适用的测试分层（逗号分隔）
     */
    @Column(name = "applicable_layers", length = 500)
    private String applicableLayers;
    
    /**
     * 适用的测试方法（逗号分隔）
     */
    @Column(name = "applicable_methods", length = 500)
    private String applicableMethods;
    
    /**
     * 适用的业务模块（逗号分隔）
     */
    @Column(name = "applicable_modules", length = 500)
    private String applicableModules;
    
    /**
     * 模板描述
     */
    @Column(name = "template_description", columnDefinition = "TEXT")
    private String templateDescription;
    
    /**
     * 版本号
     */
    @Column(name = "version")
    private Integer version = 1;
    
    /**
     * 是否启用：1-启用，0-禁用
     */
    @Column(name = "is_active", length = 1)
    private String isActive = "1";
    
    /**
     * 创建人ID
     */
    @Column(name = "creator_id")
    private Long creatorId;
    
    /**
     * 创建时间
     */
    @Column(name = "create_time")
    private LocalDateTime createTime;
    
    /**
     * 更新时间
     */
    @Column(name = "update_time")
    private LocalDateTime updateTime;
    
    @PrePersist
    public void prePersist() {
        this.createTime = LocalDateTime.now();
        this.updateTime = LocalDateTime.now();
    }
    
    @PreUpdate
    public void preUpdate() {
        this.updateTime = LocalDateTime.now();
    }
}

File: backend-python/ai-service/app/models/prompt_template.py
"""
提示词模板数据库模型
"""
from sqlalchemy import Column, BigInteger, String, Integer, Text, DateTime
from sqlalchemy.sql import func
from app.database import Base


class PromptTemplate(Base):
    """提示词模板表"""
    __tablename__ = "prompt_template"
    
    id = Column(BigInteger, primary_key=True, index=True)
    template_code = Column(String(100), unique=True, nullable=False, index=True)
    template_name = Column(String(500), nullable=False)
    template_category = Column(String(100))
    template_type = Column(String(50))
    template_content = Column(Text, nullable=False)
    template_variables = Column(Text)
    applicable_layers = Column(String(500))
    applicable_methods = Column(String(500))
    applicable_modules = Column(String(500))
    template_description = Column(Text)
    version = Column(Integer, default=1)
    is_active = Column(String(1), default="1")
    creator_id = Column(BigInteger)
    create_time = Column(DateTime(timezone=True), server_default=func.now())
    update_time = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    
    def to_dict(self):
        """转换为字典"""
        return {
            "id": self.id,
            "template_code": self.template_code,
            "template_name": self.template_name,
            "template_category": self.template_category,
            "template_type": self.template_type,
            "template_content": self.template_content,
            "template_variables": self.template_variables,
            "applicable_layers": self.applicable_layers,
            "applicable_methods": self.applicable_methods,
            "applicable_modules": self.applicable_modules,
            "template_description": self.template_description,
            "version": self.version,
            "is_active": self.is_active,
            "creator_id": self.creator_id,
            "create_time": self.create_time.isoformat() if self.create_time else None,
            "update_time": self.update_time.isoformat() if self.update_time else None,
        }

